Update the Overview section in README (#1193)
GoogleCloudPlatform/magic-modules: README.md

## Overview

> Magic Modules is a tool we use to autogenerate support in infrastructure-as-code

Reviewer: ```suggestion
Magic Modules is a tool used to autogenerate support in a variety of open source DevOps
```

tools for Google Cloud Platform. [GCP "resource"](https://cloud.google.com/docs/overview/#gcp_resources)
definitions are encoded in a shared data file, and that data is used to fill in

> tool-specific "Mad Libs"-style templates across each of the tools Magic Modules

Reviewer: ```suggestion
tool-specific templates across each of the tools Magic Modules
```

Reviewer: While I think using "mad libs" makes for a great talk at a conference, I think its a bit too campy here.

generates.

Magic Modules generates GCP support for:

* Terraform
* Ansible
* InSpec

Importantly, Magic Modules *isn't* full code generation. Every change is made

> manually; more than a code generator, Magic Modules a force multiplier for our

Reviewer: ```suggestion
manually; more than a code generator, Magic Modules is a force multiplier for 
```

> developers. While many Magic Modules resources are defined exactly based on the

Reviewer: ```suggestion
development. While many Magic Modules resources are defined exactly based on the
```
